analyzer/runtime/extract.go

```diff
@@ -578,6 +586,84 @@ func sumGasUsed(events []*EventData) (sum uint64, foundGasUsedEvent bool) {
 	return
 }
 
+// Attempts to extract the human-readable error message from
+// the raw error returned by the node.
+// If the function successfully extracts the message, it returns (msg, nil).
+// If the function cannot extract the message, it returns (nil, nil).
+// If the function encounters a malformed error message, it returns (nil, err).
+//
+// In Emerald and older Sapphire versions, transaction revert
+// reasons were decoded by the runtime and returned as
+// human-readable strings, e.g.
+// - "reverted: Incorrect premium amount."
+// - "execution failed: out of gas"
+//
+// More recent versions of Sapphire no longer decode the error
+// message and instead use the following error format:
+// - "reverted: " || base64(abiEncode(error))
+//
+// We first check to see if the error message is a plaintext error string,
+// in which case we simply sanitize and return the message.
+// Otherwise, we optimistically try to decode the error as the prevailing
+// Error(string) type and return the error message if successful.
+//
+// Note that the error type can be any data type specified in the abi.
+// We may not have the abi available now, so in those cases the
+// abi analyzer will extract the error message once the contract
+// abi is available.
+func tryParseErrorMessage(msg string) (*string, error) {
+	// Decode revert reason, format is as follows:
+	// "reverted: " || base64(abiEncode(error))
+	ErrRevertedPrefix := "reverted: "
+	if len(msg) < 10 || msg[:10] != ErrRevertedPrefix {
```

Comment:
Consider using [`strings.HasPrefix`](https://pkg.go.dev/strings#HasPrefix) since it is a bit more idiomatic go code for checking a string prefix.

Also function variables usually always start with lowercase (`errRevertedPrefix`).

Reply:
updated!